src/utils/conversion.ts

```diff
@@ -4069,7 +4069,26 @@ export function inputStatTypeToMeshStatType(
  * @hidden
  */
 export function meshProposalStatusToProposalStatus(
-  status: PalletMultisigProposalStatus
+  status: PalletMultisigProposalStatus,
+  expiry: Date | null
 ): ProposalStatus {
-  return ProposalStatus[status.type];
+  const { type } = status;
+  switch (type) {
+    case 'ActiveOrExpired':
+      if (!expiry || expiry > new Date()) {
+        return ProposalStatus.Active;
+      } else {
+        return ProposalStatus.Expired;
+      }
+    case 'Invalid':
+      return ProposalStatus.Invalid;
+    case 'ExecutionSuccessful':
```

Comment:
If you see its rare that `type` is used out of any enum. For conversion corresponding `.is<EnumValue>` is checked like `if (status.isActiveOrExpired)`

Reply:
The methods don't narrow the type down, which makes the UnrechableCaseError not work. It forces a non exhaustive match into a compile error.